Include core video return code in fatal error

low-batt opened the pull request:
This commit will add code to `VideoView` that includes the result code returned by `CVDisplayLinkCreateWithActiveCGDisplays` in the alert posted when a display link can not be created.

- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [ ] This implements/fixes issue #.

---

**Description:**

low-batt committed:
Include core video return code in fatal error

This commit will add code to VideoView that includes the result code
returned by CVDisplayLinkCreateWithActiveCGDisplays in the alert posted
when a display link can not be created.

low-batt commented:
We have not been able to determine the cause of issue #4520 where it appears the call to [CVDisplayLinkCreateWithActiveCGDisplays](https://developer.apple.com/documentation/corevideo/1456863-cvdisplaylinkcreatewithactivecgd) is failing. The commit in this pull request merely enhances the error reporting by including the return code in the fatal alert posted.

Possibly `link` should be changed to be a lazy stored property. Given there is unexplained trouble in this area I restricted the changes to just improving the error reporting.

The updated alert will look something like this if creation of the display link fails:

![display-link-alert](https://github.com/iina/iina/assets/86170219/f5367e52-634e-4505-a103-53e755bf96d9)

uiryuu commented on iina/VideoView.swift:
```diff
@@ -204,14 +204,27 @@ class VideoView: NSView {
+  /// by `Logger.fatal`.
+  /// - Returns: A [CVDisplayLink](https://developer.apple.com/documentation/corevideo/cvdisplaylink-k0k).
+  private func obtainDisplayLink() -> CVDisplayLink {
+    guard link == nil else { return link! }
```
```suggestion
    if let link = link { return link }
```

low-batt committed:
Include core video return code in fatal error

This commit will add code to VideoView that includes the result code
returned by CVDisplayLinkCreateWithActiveCGDisplays in the alert posted
when a display link can not be created.

Co-authored-by: Yuze Jiang <i@uiryuu.moe>
